crates/primitives/src/event.rs

```diff
@@ -0,0 +1,40 @@
+// Copyright 2018-2022 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use xxhash_rust::const_xxh3::xxh3_128;
+
+/// Generate the topic for the event signature.
+///
+/// xxh3_128(path) ++ xxh3_128(event_variant) todo: + fields?
+pub const fn event_signature_topic(
+    path: &'static str,
+    event_variant: &'static str,
+) -> [u8; 32] {
+    let p = xxh3_128(path.as_bytes()).to_be_bytes();
+    // todo: add fields to signature?
```

Comment:
As you mentioned in our chat, we don't need this `const fn` if we are just using `Enum::Variant`, we can do the hashing in the proc_macro itself. Currently I need this because I am using `module_path!`, but if we can do without this then we can choose our hashing algorithm.

The question remains then whether it is sufficient to use `Enum::Variant` or do we add the field names too e.g. `Enum::Variant { field_1, field_2, field_3 }` similar to Ethereum (except without type names). 

With our without the field names, need to consider the impact of signature topic collisions.